src/mugs.js

```diff
@@ -473,6 +473,15 @@ define([
             }
             return this.parentMug && this.parentMug.isInRepeat();
         },
+        /**
+         * Check if mug is referenced by other mugs
+         *
+         * @param except - Array of mugs to exclude when looking for references.
+         * @returns Boolean
+         */
+        isReferencedByOtherMugs: function (except) {
```

Comment:
Good question. I think we've bent this line all over the place because `mug.form` exists and has become the defacto way to get at the form when you have a mug. More general encapsulation best practices say rather than doing `mug.form.something(arg)` it's better to do `mug.something(arg)` and leave the fact that `mug` has a form property as an implementation detail of `mug` rather than making the calling code aware of the mug's internals.

It could be argued that this information should be passed from the form to the undo manager rather than having the undo manager reach through the mug to get it. I personally think since we reference `mug.form` all over the place, it's cleaner to ask "Is this mug referenced by other mugs?" than "Does this mug's form know if this mug is referenced by other mugs?"